src/net.cpp

```diff
@@ -1608,17 +1609,20 @@ void CConnman::ThreadDNSAddressSeed()
             if (!resolveSource.SetInternal(host)) {
                 continue;
             }
-            unsigned int nMaxIPs = 256; // Limits number of IPs learned from a DNS seed
-            if (LookupHost(host.c_str(), vIPs, nMaxIPs, true)) {
-                for (const CNetAddr& ip : vIPs) {
-                    int nOneDay = 24*3600;
-                    CAddress addr = CAddress(CService(ip, Params().GetDefaultPort()), requiredServiceBits);
-                    addr.nTime = GetTime() - 3*nOneDay - rng.randrange(4*nOneDay); // use a random age between 3 and 7 days old
-                    vAdd.push_back(addr);
-                    found++;
+            const unsigned int max_ips_per_family = 16; // Limits number of IPs learned per family from a DNS seed
+            for (int addrinfo_family : {AF_INET, AF_INET6}) {
+                if (LookupHost(host.c_str(), vIPs, max_ips_per_family, true, addrinfo_family)) {
+                    for (const CNetAddr& ip : vIPs) {
+                        int nOneDay = 24*3600;
+                        CAddress addr = CAddress(CService(ip, Params().GetDefaultPort()), requiredServiceBits);
+                        addr.nTime = GetTime() - 3*nOneDay - rng.randrange(4*nOneDay); // use a random age between 3 and 7 days old
+                        vAdd.push_back(addr);
+                        found++;
+                    }
+                    addrman.Add(vAdd, resolveSource);
```

Comment:
In the second iteration of `addrinfo_family`, `vAdd` here would also contain records from the first iteration?